lib/dialects/abstract/query-interface.js

```diff
@@ -2,25 +2,24 @@
 
 const _ = require('lodash');
 
-const Utils = require('./utils');
-const DataTypes = require('./data-types');
-const SQLiteQueryInterface = require('./dialects/sqlite/query-interface');
-const MSSQLQueryInterface = require('./dialects/mssql/query-interface');
-const MySQLQueryInterface = require('./dialects/mysql/query-interface');
-const PostgresQueryInterface = require('./dialects/postgres/query-interface');
-const Transaction = require('./transaction');
-const QueryTypes = require('./query-types');
-const Op = require('./operators');
+const Utils = require('../../utils');
+const DataTypes = require('../../data-types');
+const Transaction = require('../../transaction');
+const QueryTypes = require('../../query-types');
+const Op = require('../../operators');
 
 /**
  * The interface that Sequelize uses to talk to all databases
  *
- * @class QueryInterface
+ * @abstract
```

Comment:
This should be marked as a private class, all inherited query interface classes from dialect should be marked as 'public' by removing `@private` directive if present in their definition

Reply:
The base class isn't really private though, it's currently exposed in the docs.
If anything it should be treated like an interface but idk how to express that in jsdoc.